vars/githubscm.groovy

```diff
@@ -209,22 +209,30 @@ def hasForkPullRequest(String group, String repository, String author, String br
     return result
 }
 
-def getForkedProjectName(String group, String repository, String owner, String credentialsId = 'kie-ci1-token', int page = 1, int perPage = 100) {
+def getForkedProjectName(String group, String repository, String owner, String credentialsId = 'kie-ci1-token', int page = 1, int perPage = 100, replays = 3) {
     if (group == owner) {
         return repository;
     }
     def result = null
     withCredentials([string(credentialsId: credentialsId, variable: 'OAUTHTOKEN')]) {
-        def curlResult = sh(returnStdout: true, script: "curl -H \"Authorization: token ${OAUTHTOKEN}\" 'https://api.github.com/repos/${group}/${repository}/forks?per_page=${perPage}&page=${page}'")?.trim()
-        if (curlResult) {
-            def forkedProjects = readJSON text: curlResult
-            if (forkedProjects != null && forkedProjects.size() > 0) {
-                def forkedProject = forkedProjects.find { it.owner.login == owner }
-                result = forkedProject ? forkedProject.name : getForkedProjectName(group, repository, owner, credentialsId, ++page, perPage)
+        def forkedProjects = null
+
+        try {
+            def curlResult = sh(returnStdout: true, script: "curl -H \"Authorization: token ${OAUTHTOKEN}\" 'https://api.github.com/repos/${group}/${repository}/forks?per_page=${perPage}&page=${page}'")?.trim()
+            if (curlResult) {
+                forkedProjects = readJSON text: curlResult
+            }
+        } catch (MissingPropertyException e) {
+            if (--replays <= 0) {
```

Comment:
Couldn't we leverage https://github.com/kiegroup/jenkins-pipeline-shared-libraries/blob/master/vars/PipelineHelper.groovy ?

Reply:
or retry step ?
https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#retry-retry-the-body-up-to-n-times

Reply:
good proposal, but in our case we are trying to deal  with a problem between Jenkins and github API and our flow is doing all the checking out in one single step and we don't wan't to try the whole thing, so I guess this is the way.

Reply:
I still don't get why we would not use the `retry` command from Jenkins pipeline instead of using the `replays` variable and call again `getForkedProjectName`

Reply:
@radtriste probably I'm missing something here but I don't see the way to obtain `step` instance from https://github.com/kiegroup/jenkins-pipeline-shared-libraries/blob/5adf03cc28fa93e088651ee37997682e158aff5a/vars/githubscm.groovy#L24
could you please guide me?

So it would be to replace method calling by
```
def sourceRepository = (new PipelineHelper(STEP)).retry(getForkedProjectName(defaultAuthor, repository, sourceAuthor, credentials['token']), 3, 30)  ?: repository
```
would `STEP` be `this`?

many thanks!

Reply:
that is the idea to set `this`

Reply:
I would have done like this to leverage `PipelineHelper` =>

```
def getForkedProjectName(String group, String repository, String owner, String credentialsId = 'kie-ci1-token', int page = 1, int perPage = 100, replays = 3) {
    if (group == owner) {
        return repository;
    }
    def forkedProjects = null
    withCredentials([string(credentialsId: credentialsId, variable: 'OAUTHTOKEN')]) {
        new PipelineHelper(this)
            .retry({
                def curlResult = sh(returnStdout: true, script: "curl -H \"Authorization: token ${OAUTHTOKEN}\" 'https://api.github.com/repos/${group}/${repository}/forks?per_page=${perPage}&page=${page}'")?.trim()
                if (curlResult) {
                    forkedProjects = readJSON text: curlResult
                }
            }, 
            replays, 30, {
                println "[ERROR] Error getting forked project name for ${group}/${repository}/forks?per_page=${perPage}&page=${page}"
            })
    }
    if (forkedProjects) {
        def forkedProject = forkedProjects.find { it.owner.login == owner }
        return forkedProject ? forkedProject.name : getForkedProjectName(group, repository, owner, credentialsId, ++page, perPage)
    }
    return null
}
```

just a proposition